File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix_list = []
        suffix_list = []
        list_to_return = []

        for i in range(len(nums)):
            if(i == 0):
                prefix_list.append(nums[i])
            else:
                prefix_list.append(prefix_list[i-1] * nums[i])

        for i in range(len(nums) - 1, -1, -1):
            if(i == (len(nums) - 1)):
                suffix_list.append(nums[i])
            else:
                suffix_list.insert(0, suffix_list[0]*nums[i])

        for i in range(len(nums)):
            if(i == 0):
                list_to_return.append(suffix_list[i+1])
            elif(i == (len(nums) - 1)):
                list_to_return.append(prefix_list[i-1])
            else:
                list_to_return.append(prefix_list[i-1]*suffix_list[i+1])
        
        return list_to_return

File: Data Structures & Algorithms/three-integer-sum/submission-0.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums) {
        var res = new List<List<int>>();
        Array.Sort(nums);

        for(int i = 0; i < nums.Length; i++) {
            if (i > 0 && nums[i] == nums[i - 1]) {
                continue;
            }

            int l = i + 1;
            int r = nums.Length - 1;

            while (l < r) {
                int threeSum = nums[i] + nums[l] + nums[r];
                if (threeSum > 0) {
                    r--;
                }
                else if (threeSum < 0) {
                    l++;
                }
                else {
                    res.Add(new List<int> {nums[i], nums[l], nums[r]});
                    l++;
                    while (nums[l] == nums[l - 1] && l < r) {
                        l++;
                    }
                }
            }
        }
        return res;
    }
}
